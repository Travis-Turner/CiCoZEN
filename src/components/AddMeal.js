import React from 'react';
import { connect } from 'react-redux'
import { startAddMeal } from '../actions/meals';

export class AddMeal extends React.Component {
    constructor(props){
        super(props);
        this.state = {
            error: ''
        }
    }
    handleSubmit = (e) => {
        e.preventDefault();
        const name = e.target.elements[0].value;
        const calories = e.target.elements[1].value;
        const carbohydrates = e.target.elements[2].value;
        const protein = e.target.elements[3].value;
        const fat = e.target.elements[4].value;
        //Form validation
        if (!name){
            return this.setState(() => ({
                error: 'Please enter a name for your meal.'
            }));
        } 
        if (!calories){
            return this.setState(() => ({
                error: 'You must provide the number of calories in your meal.'
            }))
        }
        //Add meal
        const meal = {
            name,
            calories,
            carbohydrates,
            protein,
            fat
        }
        this.props.startAddMeal(meal); 
        this.props.history.push('/');
    }
    render(){
        return (
            <div>
                <form onSubmit={this.handleSubmit}>
                    {this.state.error && <p>{this.state.error}</p>}
                    <p>Name:</p>
                    <input name='name' type='text' />
                    <p>Calories:</p>
                    <input name='calories' type='number' />
                    <p>Carbohydrates:</p>
                    <input name='carbohydrates' type='number' />
                    <p>Protein:</p>
                    <input name='protein' type='number' />
                    <p>Fat:</p>
                    <input name='fat' type='number' />
                    <button>Add Meal</button>
                </form>
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => ({
    startAddMeal: (meal) => dispatch(startAddMeal(meal))
  });
  
  export default connect (undefined, mapDispatchToProps)(AddMeal);