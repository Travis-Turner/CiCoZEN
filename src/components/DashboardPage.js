import React from 'react';
import { connect } from 'react-redux';
import { SingleDatePicker } from 'react-dates';
import moment from 'moment';
import { startSetCurrentDate } from '../actions/date';
import { startSetTotals } from '../actions/dailyTotals';
import { Link } from 'react-router-dom';
import getDailyTotals from '../selectors/getDailyTotals';

export class DashboardPage extends React.Component{
  constructor(props){
    super(props);
    this.state = {
      date: moment(this.props.currentDate),
      focused: false
    }
  }
  onDateChange = (date) => {
    const newDate = date;
    this.setState(() => ({ date: newDate }));
    this.props.startSetCurrentDate(newDate).then(() => {
      this.props.startSetTotals()
    });
  }
  onFocusChange = ({ focused }) => {
    this.setState(() => ({ focused }));
  }
  render (){
    return (
      <div>
        <Link to="/addMeal">
          Add Meal
        </Link>
        <SingleDatePicker
          date={this.state.date}
          onDateChange={this.onDateChange}
          focused={this.state.focused}
          onFocusChange={this.onFocusChange}
          isOutsideRange={() => false}
          numberOfMonths={1}
          daySize={50}
          withPortal={true}
        />
        <p>CURRENT CALORIES - {this.props.dailyTotals.calories} </p>
        <p>GOAL CALORIES - {this.props.goal}</p>
      </div>
    );
  }
} 

const mapStateToProps = (state) => {
  const dailyTotals = getDailyTotals(state.meals);
  return {
    goal: state.userSettings.goal,
    currentDate: state.currentDate,
    dailyTotals
  }
};

const mapDispatchToProps = (dispatch) => ({
  startSetCurrentDate: (date) => dispatch(startSetCurrentDate(date)),
  startSetTotals: () => dispatch(startSetTotals())
});

export default connect (mapStateToProps, mapDispatchToProps)(DashboardPage);
