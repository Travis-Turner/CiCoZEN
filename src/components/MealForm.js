import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';

export default class MealForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      name: props.meal ? props.meal.name : '',
      calories: props.meal ? props.meal.calories : 0,
      protein: props.meal ? props.meal.protein : 0,
      carbohydrates: props.carbohydrates ? props.meal.carbohydrates : 0,
      fat: props.fat ? props.meal.fat : 0,
      createdAt: props.meal ? moment(props.meal.createdAt) : moment(),
      calendarFocused: false,
      error: ''
    };
  }
  onDescriptionChange = (e) => {
    const name = e.target.value;
    this.setState(() => ({ name }));
  };
  onCaloriesChange = (e) => {
    const calories = e.target.value;

    if (!calories || calories.match(/^\d{1,}(\.\d{0,2})?$/)) {
      this.setState(() => ({ calories }));
    }
  };
  onDateChange = (createdAt) => {
    if (createdAt) {
      this.setState(() => ({ createdAt }));
    }
  };
  onFocusChange = ({ focused }) => {
    this.setState(() => ({ calendarFocused: focused }));
  };
  onSubmit = (e) => {
    e.preventDefault();

    if (!this.state.name || !this.state.calories) {
      this.setState(() => ({ error: 'Please provide food name and calorie amount.' }));
    } else {
      this.setState(() => ({ error: '' }));
      this.props.onSubmit({
        name: this.state.name,
        calories: this.state.calories,
        createdAt: this.state.createdAt.valueOf()
      });
    }
  };
  render() {
    return (
      <div>
        {this.state.error && <p>{this.state.error}</p>}
        <form onSubmit={this.onSubmit}>
          <input
            type="text"
            placeholder="Description"
            autoFocus
            value={this.state.description}
            onChange={this.onDescriptionChange}
          />
          <input
            type="text"
            placeholder="Calories"
            value={this.state.calories}
            onChange={this.onCaloriesChange}
          />
          <SingleDatePicker
            date={this.state.createdAt}
            onDateChange={this.onDateChange}
            focused={this.state.calendarFocused}
            onFocusChange={this.onFocusChange}
            numberOfMonths={1}
            isOutsideRange={() => false}
          />
          <textarea
            placeholder="Add a note for your meal (optional)"
            value={this.state.note}
            onChange={this.onNoteChange}
          >
          </textarea>
          <button>Add Meal</button>
        </form>
      </div>
    )
  }
}