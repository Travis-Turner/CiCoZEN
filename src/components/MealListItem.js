import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';

const MealListItem = ({ id, name, calories, createdAt }) => (
  <div>
    <Link to={`/edit/${id}`}>
      <h3>{name}</h3>
    </Link>
    <p>
      {calories} calories.
      -
      {moment(createdAt).format('MMMM Do, YYYY')}
    </p>
  </div>
);

export default MealListItem;
