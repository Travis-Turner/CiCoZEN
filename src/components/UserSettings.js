import React from 'react';
import { connect } from 'react-redux';
import { startUpdateGoal } from '../actions/userSettings';

export class UserSettings extends React.Component {
    constructor(props){
        super(props);
        this.state = {
            goalCalories: 0
        }
    }
    onGoalChange = (e) => {
        const goalCalories = e.target.value;
        this.setState(() => ({ goalCalories }));
    }
    handleFormSubmit = (e) => {
        e.preventDefault();
        this.props.startUpdateGoal(this.state.goalCalories);
        this.props.history.push('/dashboard');
    }
    render () {
        return (
            <div>
                <form onSubmit={this.handleFormSubmit}>
                    <input 
                        name='goalCalories' 
                        type='text' 
                        placeholder='Goal calories' 
                        onChange={this.onGoalChange}
                    />
                    <button>Submit</button>
                </form>
            </div>
        )
    }
}

const mapStateToProps = (state) => ({
    goalCalories: state.goalCalories
  });

const mapDispatchToProps = (dispatch) => ({
    startUpdateGoal: (goalCalories) => dispatch(startUpdateGoal(goalCalories))
  });

export default connect (mapStateToProps, mapDispatchToProps)(UserSettings);