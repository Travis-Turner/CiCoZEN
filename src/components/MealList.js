import React from 'react';
import { connect } from 'react-redux';
import MealListItem from './MealListItem';
import selectMeals from '../selectors/meals';

export const MealList = (props) => (
  <div>
    {
      props.meals.length === 0 ? (
        <p>No meals</p>
      ) : (
          props.meals.map((meal) => {
            return <MealListItem key={meal.id} {...meal} />;
          })
        )
    }
  </div>
);

const mapStateToProps = (state) => {
  return {
    meals: selectMeals(state.meals)
  };
};

export default connect(mapStateToProps)(MealList);