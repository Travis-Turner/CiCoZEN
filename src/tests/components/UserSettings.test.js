import React from 'react';
import { shallow } from 'enzyme';
import { UserSettings } from '../../components/UserSettings';


test('should render user settings page correctly', () => {
    const wrapper = shallow(<UserSettings />);
    expect(wrapper).toMatchSnapshot();
});