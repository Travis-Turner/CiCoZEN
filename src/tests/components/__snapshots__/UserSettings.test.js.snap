// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render user settings page correctly 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      name="goalCalories"
      onChange={[Function]}
      placeholder="Goal calories"
      type="text"
    />
    <button>
      Submit
    </button>
  </form>
</div>
`;
